From ed3f4d7943d3cfc3d639102afbb443f23a0734bc Mon Sep 17 00:00:00 2001
From: Rui Loura <rui@joyent.com>
Date: Mon, 8 Jan 2018 19:45:02 +0000
Subject: [PATCH] NET-384 sdc-booter should dump core on uncaught exception
 NET-385 sdc-booter panics while handling discover on log level trace

---
 bin/dhcpd    |  2 +-
 lib/dhcpd.js | 13 +++++++++----
 2 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/bin/dhcpd b/bin/dhcpd
index 3e21437..8822d6e 100755
--- a/bin/dhcpd
+++ b/bin/dhcpd
@@ -16,4 +16,4 @@ if [[ ! -d "${NODE_INSTALL}" && -d "${TOP}/build/node" ]]; then
     NODE_INSTALL=${TOP}/build/node
 fi
 
-${NODE_INSTALL}/bin/node ${TOP}/server.js "$@"
+${NODE_INSTALL}/bin/node --abort-on-uncaught-exception ${TOP}/server.js "$@"
diff --git a/lib/dhcpd.js b/lib/dhcpd.js
index fd974ae..ea33deb 100755
--- a/lib/dhcpd.js
+++ b/lib/dhcpd.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2016 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -74,10 +74,15 @@ DHCPD.prototype.handleMessage = function (msg, peer) {
 
     // Print the whole packet in hex
     if (log.trace()) {
+        var str_buf;
         for (var i = 0; i < msg.length; i += 4) {
-            // XXX
-            this.log.trace(sprintf('[%03d]: 0x%02x 0x%02x 0x%02x 0x%02x',
-                i, msg[i], msg[i+1], msg[i+2], msg[i+3]));
+            str_buf = '';
+            str_buf += sprintf('[%03d]:', i);
+
+            for (var j = 0; j < Math.min(4, msg.length - i); j++) {
+                str_buf += sprintf(' 0x%02x', msg[i + j]);
+            }
+            this.log.trace(str_buf);
         }
     }
 
-- 
2.21.0

